baselines/FedMLB/README.md

```diff
@@ -0,0 +1,588 @@
+
+`local_batch_size = round(local_examples * local_epochs / local_updates)`
+
+This is the default configuration also in this repository.
+
+To override this behaviour, and set a specific local batch size (local epochs will still 
+be valid), use the following argument when launching the simulation.
+
+```
+# this will set the local batch size to 64
+python -m FedMLB.main batch_size=64
+```
+
+Alternatively, it can be done by directly modifying the `base.yaml` config file by changing the value
+of `batch_size`, as follows:
+```
+# base.yaml
+...
+batch_size=64 # default to null
+...
+``` 
+
+
+### Splitting a Simulation in a bunch of Shorter Simulations
+`simulation_manager.sh` contains a simple bash script to
+divide a simulation in a bunch of shorter simulations.
+For example, it can divide a simulation of 1000 rounds 
+(`total_round`) in
+50 consecutive simulations of 20 rounds each (`rounds_per_run`).
+By the automatic retrieve of last checkpoint, 
+the current simulation
+restarts from where the previous one stopped and saves a
+checkpoint at the end.
+
+To be able to run the script, first give it the execution permission
+(for Linux machines) with:
+``` 
+chmod +x ./simulation_manager.sh
+``` 
+Then launch it with:
+``` 
+./simulation_manager.sh
+``` 
+## Expected Results
+This repository can reproduce the results for 3 baselines used in the experimental part
+of the original paper: FedMLB, FedAvg, FedAvg+KD.
+
+The following tables compares results produced via the code in this repository
+with the results reported in the paper.
+Results from the paper are reported in brackets. 
+Note that (as in the original paper),
+the accuracy at the target round is based on the exponential moving average with the momentum parameter
+0.9. Tensorboard allows to easily visualize/calculate the moving average of a run.
+
+All the results reported in this repository can be also browsed 
+on this [tesnsorboard.dev online dashboard](https://tensorboard.dev/experiment/n5QWJWP8QWagUxcWuOtAUA/#scalars).
+
+### Table 1a
+
+The results of Table 1a in the paper (for FedAvg and FedMLB) refers
+to CIFAR-100 (Figure 3) and Tiny-ImageNet (Figure 7a) with 
+the setting (1.) Moderate-scale with Dir(0.3), 100 clients, 5% participation rate.
+
+To reproduce the results run the following:
+
+```bash
+# this will produce six consecutive runs
+python -m FedMLB.main --multirun dataset_config.dataset="cifar100","tiny-imagenet" algorithm="FedMLB","FedAvg","FedAvg+KD" 
+```
+#### CIFAR-100, Dir(0.3), 100 clients, 5% participation.
+
+<table>
+  <tr>
+    <td></td>
+    <td colspan="2">Accuracy @500R</td>
+    <td colspan="2">Accuracy @1000R</td>
+  </tr>
+  <tr> 
+    <td>Method</td>
+    <td>Paper</td>
+    <td>This repo</td>
+    <td>Paper</td>
+    <td>This repo</td>
+  </tr>
+  <tr>
+    <td>FedAvg</td>
+    <td>41.88</td>
+    <td>44.52</td>
+    <td>47.83</td>
+    <td>49.15</td>
+  </tr>
+  <tr>
+    <td>FedAvg+KD</td>
```

Comment:
I don't see `FedAvg + KD` in Table 1a in the paper

Reply:
You are right. Actually, the results for ` FedAvg + KD` are reported in Table 2 in the paper.